build/README.md

```diff
@@ -417,7 +417,7 @@ command from within `make shell` or locally, to refresh your authentication toke
 The version of this build. Version defaults to the short hash of the latest commit.
 
 ### REGISTRY
-The registry that is being used to store docker images. Defaults to gcr.io/agones-images - the release + CI registry.
+The registry that is being used to store docker images. Defaults to us-docker.pkg.dev/agones-images/release
```

Comment:
Since we can now split out registries, I'm wondering if the default (e.g. if you forget to set the `REGISTRY` environment variable) should be something else to make it harder to accidentally push to the release registry. 

Should the registry default to something like `us-docker.pkg.dev/agones-images/dev`? And be overridden in the `make do-release` target to point to the release registry?

Ideally we'd have cloud build (or some similar build automation) creating our releases and restrict access to the release registry to only automation.

Reply:
The `us-docker.pkg.dev/agones-images/release` registry only give `allUsers` reader access so `make push` won't succeed without override the `REGISTRY`.
Do we want to allow users to push image to any of the registry under `agones-images` project? Should we just enforce users to use their own registry for development?
In our current setting, the `gcr.io` registry has public visibility. Does it mean arbitrary users can push images to it? I understand it gives all users pull permission, put didn't find any information about push in the public docs.

Reply:
> `-docker.pkg.dev/agones-images/release` registry only give `allUsers` reader access so `make push` won't succeed without override the `REGISTRY`.

But for maintainers that have editor access on the project, we can still push to it (so we can cut releases) but this means we can also inadvertently push to it. I've done this in the past with the GCR bucket where I forgot to set the `REGISTRY` variable before running a push command and accidentally pushed into our official bucket (luckily my build looked like any other CI build so it didn't cause any problems). But now that CI is split out from releases, if I accidentally pushed it would put an extraneous build into the releases bucket. 

> Do we want to allow users to push image to any of the registry under agones-images project? Should we just enforce users to use their own registry for development?

Correct. Users should use their own. But as I mentioned above for folks that have access to the agones-images project so we can push official builds we can accidentally clobber official releases.

Reply:
Oh, I see your point. Didn't realize the editors have all the access. Makes perfect sense!

Reply:
I like the idea of setting it to something default that isn't a real registry.

A couple of suggestions:
1. Make it blank by default. So images are local only unless you specify a REGISTRY.
2. Make it "agones-images", so while it doesn't exist, it does group all Agones with a prefix.

I did check, and all images start wtih `agones-`, so I'd be tempted to lean towards No. 1, but No. 2 may be safer over time, just because of the extra prefix.

Reply:
If we leave it as blank or only make it a prefix, the docker commands (such as in `make push` `make install`) will default it as a docker registry, which leading to more confusion I think.

Reply:
🤔 why do you think it will create more confusion? Either way, it still ends up failing if someone attempts to push the image.

But to that point - that would say to me that leaving is blank is actually the best option to indicate "this is only a local version, not a release version you've got in your local images".

If I run `docker images` and I see the following:

```shell
❯ docker images
REPOSITORY                TAG                          IMAGE ID       CREATED              SIZE
agones-allocator          1.29.0-c61a21a-amd64         8df912cba403   About a minute ago   68.4MB
agones-ping               1.29.0-c61a21a-amd64         ee443e7013f0   About a minute ago   24.4MB
agones-extensions         1.29.0-c61a21a-amd64         a38e43db5912   2 minutes ago        69.4MB
agones-controller         1.29.0-c61a21a-amd64         55e5cd993a40   2 minutes ago        69.4MB
agones-sdk                1.29.0-c61a21a-linux-amd64   8a0612f7cfe4   3 minutes ago        65.3MB
agones-build              f1d1057367                   cd6bfe544e90   4 weeks ago          3.57GB
agones-build-sdk-node     2a5f9e6f1a                   6967d5d17df4   7 weeks ago          1.83GB
agones-build-sdk-base     4e48d2360c                   5348986fd7ce   7 weeks ago          1.59GB
debian                    bookworm                     0ac8b312578d   5 months ago         119MB
```

I think that's fairly obvious that it isn't coming from a Docker Hub -- but what do you think?

Also, if we do set a prefix of `agones-images`  - then someone could create a user of `agones-images` on Docker Hub, and I think that _would_ definitely be confusing 😄 

Just to see it side by side with the other suggestion:

```
❯ docker images
REPOSITORY                TAG                          IMAGE ID       CREATED              SIZE
agones-images/agones-allocator          1.29.0-c61a21a-amd64         8df912cba403   About a minute ago   68.4MB
agones-images/agones-ping               1.29.0-c61a21a-amd64         ee443e7013f0   About a minute ago   24.4MB
agones-images/agones-extensions         1.29.0-c61a21a-amd64         a38e43db5912   2 minutes ago        69.4MB
agones-images/agones-controller         1.29.0-c61a21a-amd64         55e5cd993a40   2 minutes ago        69.4MB
agones-images/agones-sdk                1.29.0-c61a21a-linux-amd64   8a0612f7cfe4   3 minutes ago        65.3MB
agones-build                            f1d1057367                   cd6bfe544e90   4 weeks ago          3.57GB
agones-build-sdk-node                   2a5f9e6f1a                   6967d5d17df4   7 weeks ago          1.83GB
agones-build-sdk-base                   4e48d2360c                   5348986fd7ce   7 weeks ago          1.59GB
debian                                  bookworm                     0ac8b312578d   5 months ago         119MB
```

Eh - actually looking at it. I don't mind either 🤷🏻

Reply:
I had the impression that for the maintainers, we'd like to provide a valid default registry for the development with overriding the environment viable. Otherwise, there was no reason to have this [`release_registry`](https://github.com/googleforgames/agones/blob/main/build/Makefile#L30) defined for the development process at first place, i.e. we could separate it from the registry which is used for the real release (`make do-release`) and make it default to local at the beginning. It has nothing to do with the limitation of GCR that we cannot split out registries.

But I agreed with you, seems like even for the maintainers, we should prevent them to use `agones-images` project for development purpose. And make it a hard requirement to set the `REGISTRY` environment viable for local development that follows the guidance is reasonable.

Reply:
I removed `release_registry` from the `Makefile` and add instructions to indicates it's required to set this environment viable to follow the guidance to build, push, and install Agones for dev purpose.